[Test] consistent CI failure for a test related to ES

**Issue**

CI is consistently failing this test.

For example, see https://github.com/kanisterio/kanister/actions/runs/3492385063/jobs/5846995089

```
----------------------------------------------------------------------
FAIL: <autogenerated>:1: Elasticsearch.TestRun

integration_test.go:239:
    c.Assert(err, IsNil)
... value *errors.withStack = Context done while polling: context deadline exceeded ("Context done while polling: context deadline exceeded")

{"File":"pkg/log/log.go","Function":"github.com/kanisterio/kanister/pkg/log.Print","Line":165,"app":"elasticsearch","hostname":"fv-az365-424","level":"info","msg":"UnInstalling the application using helm.","time":"2022-11-17T23:33:28.219947233Z"}
OOPS: 5 passed, 1 FAILED
--- FAIL: Test (264.44s)
```


**To Reproduce**

Submit PR that triggers tests execution.

**Expected behavior**

For tests to succeed and CI to pass.


**Environment**

GH CI

**Additional context**

https://github.com/kanisterio/kanister/actions/runs/3492385063/jobs/5846995089

--- a/examples/elasticsearch/elasticsearch-blueprint.yaml
+++ b/examples/elasticsearch/elasticsearch-blueprint.yaml
@@ -11,6 +11,11 @@ actions:
     phases:
     - func: KubeTask
       name: backupToObjectStore
+      objects:
+        esMasterCredSecret:
+          kind: Secret
+          name: "{{ index .Object.metadata.labels.app }}-credentials"
+          namespace: "{{ .StatefulSet.Namespace }}"
       args:
         namespace: "{{ .StatefulSet.Namespace }}"
         image: "ghcr.io/kanisterio/es-sidecar:0.84.0"
@@ -24,7 +29,9 @@ actions:
         - |
           host_name="{{ .Object.spec.serviceName }}.{{ .StatefulSet.Namespace }}.svc.cluster.local"
           BACKUP_LOCATION=es_backups/{{ .StatefulSet.Namespace }}/{{ .StatefulSet.Name }}/{{ toDate "2006-01-02T15:04:05.999999999Z07:00" .Time | date "2006-01-02T15:04:05Z07:00" }}/backup.gz
-          elasticdump --bulk=true --input=http://${host_name}:9200 --output=/backup
+          master_username="{{ index .Phases.backupToObjectStore.Secrets.esMasterCredSecret.Data "username" | toString }}"
+          master_password="{{ index .Phases.backupToObjectStore.Secrets.esMasterCredSecret.Data "password" | toString }}"
+          NODE_TLS_REJECT_UNAUTHORIZED=0 elasticdump --bulk=true --input=https://${master_username}:${master_password}@${host_name}:9200 --output=/backup
           gzip /backup
           kando location push --profile '{{ toJson .Profile }}'  /backup.gz --path $BACKUP_LOCATION
           kando output backupLocation $BACKUP_LOCATION
@@ -34,6 +41,11 @@ actions:
     phases:
     - func: KubeTask
       name: restoreFromObjectStore
+      objects:
+        esMasterCredSecret:
+          kind: Secret
+          name: "{{ index .Object.metadata.labels.app }}-credentials"
+          namespace: "{{ .StatefulSet.Namespace }}"
       args:
         namespace: "{{ .StatefulSet.Namespace }}"
         image: "ghcr.io/kanisterio/es-sidecar:0.84.0"
@@ -46,7 +58,9 @@ actions:
         - -c
         - |
           host_name="{{ .Object.spec.serviceName }}.{{ .StatefulSet.Namespace }}.svc.cluster.local"
-          kando location pull --profile '{{ toJson .Profile }}' --path '{{ .ArtifactsIn.cloudObject.KeyValue.backupLocation }}' - | gunzip | elasticdump --bulk=true --input=$ --output=http://${host_name}:9200
+          master_username="{{ index .Phases.restoreFromObjectStore.Secrets.esMasterCredSecret.Data "username" | toString }}"
+          master_password="{{ index .Phases.restoreFromObjectStore.Secrets.esMasterCredSecret.Data "password" | toString }}"
+          kando location pull --profile '{{ toJson .Profile }}' --path '{{ .ArtifactsIn.cloudObject.KeyValue.backupLocation }}' - | gunzip | NODE_TLS_REJECT_UNAUTHORIZED=0 elasticdump --bulk=true --input=$ --output=https://${master_username}:${master_password}@${host_name}:9200
   delete:
     inputArtifactNames:
     - cloudObject

--- a/pkg/app/elasticsearch.go
+++ b/pkg/app/elasticsearch.go
@@ -55,7 +55,7 @@ type ElasticsearchInstance struct {
 	elasticsearchURL string
 }
 
-// Last tested on 7.10.1
+// Last tested on 8.5.1
 func NewElasticsearchInstance(name string) App {
 	return &ElasticsearchInstance{
 		name:      name,
@@ -71,7 +71,7 @@ func NewElasticsearchInstance(name string) App {
 				"replicas":     "1",
 			},
 		},
-		elasticsearchURL: "localhost:9200",
+		elasticsearchURL: "https://localhost:9200",
 	}
 }
 
@@ -144,6 +144,16 @@ func (esi *ElasticsearchInstance) Uninstall(ctx context.Context) error {
 	return nil
 }
 
+func (esi *ElasticsearchInstance) Secrets() map[string]crv1alpha1.ObjectReference {
+	return map[string]crv1alpha1.ObjectReference{
+		"elasticsearch": crv1alpha1.ObjectReference{
+			Kind:      "Secret",
+			Name:      esi.chart.Chart + "-master-credentials",
+			Namespace: esi.namespace,
+		},
+	}
+}
+
 func (esi *ElasticsearchInstance) GetClusterScopedResources(ctx context.Context) []crv1alpha1.ObjectReference {
 	return nil
 }
@@ -151,7 +161,7 @@ func (esi *ElasticsearchInstance) GetClusterScopedResources(ctx context.Context)
 func (esi *ElasticsearchInstance) Ping(ctx context.Context) error {
 	log.Print("Pinging the application to check if its accessible.", field.M{"app": esi.name})
 
-	pingCMD := []string{"sh", "-c", fmt.Sprintf("curl %s", esi.elasticsearchURL)}
+	pingCMD := []string{"sh", "-c", esi.curlCommand("GET", "")}
 	_, stderr, err := esi.execCommand(ctx, pingCMD)
 	if err != nil {
 		return errors.Wrapf(err, "Failed to ping the application. Error:%s", stderr)
@@ -161,7 +171,7 @@ func (esi *ElasticsearchInstance) Ping(ctx context.Context) error {
 	return nil
 }
 func (esi *ElasticsearchInstance) Insert(ctx context.Context) error {
-	addDocumentToIndexCMD := []string{"sh", "-c", fmt.Sprintf("curl -X POST %s/%s/_doc/?refresh=true -H 'Content-Type: application/json' -d'{\"appname\": \"kanister\" }'", esi.elasticsearchURL, esi.indexname)}
+	addDocumentToIndexCMD := []string{"sh", "-c", esi.curlCommandWithPayload("POST", esi.indexname+"/_doc/?refresh=true", "'{\"appname\": \"kanister\" }'")}
 	_, stderr, err := esi.execCommand(ctx, addDocumentToIndexCMD)
 	if err != nil {
 		// even one insert failed we will have to return becasue
@@ -174,7 +184,7 @@ func (esi *ElasticsearchInstance) Insert(ctx context.Context) error {
 }
 
 func (esi *ElasticsearchInstance) Count(ctx context.Context) (int, error) {
-	documentCountCMD := []string{"sh", "-c", fmt.Sprintf("curl %s/%s/_search?pretty", esi.elasticsearchURL, esi.indexname)}
+	documentCountCMD := []string{"sh", "-c", esi.curlCommand("GET", esi.indexname+"/_search?pretty")}
 	stdout, stderr, err := esi.execCommand(ctx, documentCountCMD)
 	if err != nil {
 		return 0, errors.Wrapf(err, "Error %s Counting the documents in an index.", stderr)
@@ -195,7 +205,7 @@ func (esi *ElasticsearchInstance) Reset(ctx context.Context) error {
 	log.Print("Resetting the application.", field.M{"app": esi.name})
 
 	// delete the index and then create it, in order to reset the es application
-	deleteIndexCMD := []string{"sh", "-c", fmt.Sprintf("curl -X DELETE %s/%s?pretty", esi.elasticsearchURL, esi.indexname)}
+	deleteIndexCMD := []string{"sh", "-c", esi.curlCommand("DELETE", esi.indexname+"/?pretty")}
 	_, stderr, err := esi.execCommand(ctx, deleteIndexCMD)
 	if err != nil {
 		return errors.Wrapf(err, "Error %s while deleting the index %s to reset the application.", stderr, esi.indexname)
@@ -207,7 +217,7 @@ func (esi *ElasticsearchInstance) Reset(ctx context.Context) error {
 // Initialize is used to initialize the database or create schema
 func (esi *ElasticsearchInstance) Initialize(ctx context.Context) error {
 	// create the index
-	createIndexCMD := []string{"sh", "-c", fmt.Sprintf("curl -X PUT %s/%s?pretty", esi.elasticsearchURL, esi.indexname)}
+	createIndexCMD := []string{"sh", "-c", esi.curlCommand("PUT", esi.indexname+"/?pretty")}
 	_, stderr, err := esi.execCommand(ctx, createIndexCMD)
 	if err != nil {
 		return errors.Wrapf(err, "Error %s: Resetting the application.", stderr)
@@ -221,4 +231,12 @@ func (esi *ElasticsearchInstance) execCommand(ctx context.Context, command []str
 		return "", "", errors.Wrapf(err, "Error getting the pod and container name %s.", esi.name)
 	}
 	return kube.Exec(esi.cli, esi.namespace, podname, containername, command, nil)
+}
+
+func (esi *ElasticsearchInstance) curlCommand(method, path string) string {
+	return fmt.Sprintf("curl -k -X %s -H 'Content-Type: application/json' -u elastic:${ELASTIC_PASSWORD} %s/%s", method, esi.elasticsearchURL, path)
+}
+
+func (esi *ElasticsearchInstance) curlCommandWithPayload(method, path, data string) string {
+	return fmt.Sprintf("%s -d %s", esi.curlCommand(method, path), data)
 }

